pkg/ipam/multipool.go

```diff
@@ -239,14 +240,32 @@ func newMultiPoolManager(conf Configuration, nodeWatcher nodeWatcher, owner Owne
 	}
 
 	// Subscribe to CiliumNode updates
-	nodeWatcher.RegisterCiliumNodeSubscriber(c)
+	c.eventloop(nodes)
 	owner.UpdateCiliumNodeResource()
 
 	c.waitForAllPools()
 
 	return c
 }
 
+func (m *multiPoolManager) eventloop(res agentK8s.LocalCiliumNodeResource) {
+	// Don't retry events - the downstream code isn't setup to handle retries.
+	// Ensure that the subscrption happens synchronously (but process async), so
+	// that there's no window in which we may miss events.
+	evs := res.Events(context.TODO(), resource.WithErrorHandler(resource.RetryUpTo(0)))
```

Comment:
Do we have a proper context to use here that is cancelled when IPAM stops? This might be dangerous as Resource[T] might be emitting events while systems that this is calling into are being stopped.

Reply:
Thanks!